### PR TITLE
Document returned entity

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: plugins/modules/foreman_architecture.py [hunk 1]
```diff
@@ -82,7 +82,12 @@
     state: absent
 '''
 
-RETURN = ''' # '''
+RETURN = '''
+entity:
+  description: Final state of the affected Entity
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]:
```suggestion
  description: Final state of the affected entity.
```

File: plugins/modules/foreman_architecture.py [hunk 2]
```diff
@@ -82,7 +82,12 @@
...
+  returned: success
+  type: dict
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]: given it is really a dict with arrays of dicts, can we be more verbose here?
also see https://github.com/theforeman/foreman-ansible-modules/issues/583

Comment (mdellweg) [reply plugins/modules/foreman_architecture.py]: Do you mean that level of 'more verbose'?

Comment (evgeni) [reply plugins/modules/foreman_architecture.py]: Yes. That looks good. Ans now you can't just copy&paste this to all modules :P

File: plugins/modules/foreman_architecture.py [hunk 3]
```diff
@@ -82,7 +82,26 @@
     state: absent
 '''
 
-RETURN = ''' # '''
+RETURN = '''
+entity:
+  description: Final state of the affected entities grouped by their type
```
Comment (evgeni) [inline plugins/modules/foreman_architecture.py]:
```suggestion
  description: Final state of the affected entities grouped by their type.
```

**Comment by evgeni** [comment]
I like!

File: plugins/modules/foreman_auth_source_ldap.py [hunk 4]
```diff
@@ -160,7 +160,108 @@
...
+        attr_mail": "mail",
+        attr_photo": null,
+        base_dn": "uid",
+        external_usergroups": [],
+        groups_base": "cn=groups,cn=accounts,dc=example,dc=com",
+        ldap_filter": "(uid=%s)",
+        onthefly_register": true,
+        type": "AuthSourceLdap",
+        use_netgroups": false,
+        usergroup_sync": true
```
Comment (mdellweg) [inline plugins/modules/foreman_auth_source_ldap.py]: I don't even know what those are for.
Anyone keen to use the suggests feature?

**Pull request closed without merging**